packages/sdk/tests/local/test_local_training.py

```diff
@@ -0,0 +1,117 @@
+#  Copyright 2020 EPAM Systems
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import os
+from typing import Optional
+from unittest.mock import ANY
+
+import pytest
+from pytest_mock import MockFixture
+
+from odahuflow.sdk.local import training
+from odahuflow.sdk.local.training import compile_artifact_name_template, TemplateContext, start_train, \
+    DEFAULT_MODEL_DIR_TEMPLATE
+from odahuflow.sdk.models import K8sTrainer, ModelTraining, ModelTrainingSpec, ModelIdentity
+
+
+# Format: [(template, context, expected_result), ...]
+test_data = [
+    (
+        '{{ .Name }}_abc_{{ .RandomUUID }}',
+        TemplateContext(Name='wine', Version='2.3', RandomUUID='123-123'),
+        'wine_abc_123-123'
+    ),
+    (
+        '{{ .Name }}-{{ .Version }}-{{ .RandomUUID }}',
+        TemplateContext(Name='model_name', Version='99', RandomUUID='uuid_here'),
+        'model_name-99-uuid_here'
+    ),
+    (
+        'Model {{ .Name }}',
+        TemplateContext(Name=None, Version=None, RandomUUID=None),
+        'Model None'
+    )
```

Comment:
```suggestion
    (
        'Model {{ .Name       }}, {{      .Version      }} ',
        TemplateContext(Name=None, Version=None, RandomUUID=None),
        'Model None, None '
    )
```
check for whitespaces?

Reply:
OK, I added whitespaces to one of tests.
I intentionally left this one without `.Version` so that there is a mismatch in number of parameters in template and in context. 
I would leave it as is. OK with you?

Reply:
agree